packages/types/src/client.ts

```diff
@@ -108,6 +108,14 @@ export interface Client<O extends ClientOptions = ClientOptions> {
   /** Returns the client's instance of the given integration class, it any. */
   getIntegration<T extends Integration>(integration: IntegrationClass<T>): T | null;
 
+  /**
+   * Add an integration to the client.
+   * This can be used to e.g. lazy load integrations.
+   * In most cases, this should not be necessary, and you're better off just passing the integrations via `integrations: []` at initialization time.
+   * However, if you find the need to conditionally load & add an integration, you can use `addIntegration` to do so.
+   * */
+  addIntegration?(integration: Integration): void;
```

Comment:
I think we want to make this method required eventually, so let's add a TODO here for v8?

Reply:
Good point! 👍

Reply:
@mydea 
Hello, I have a question about `addIntegration` method when I use [Replay functionality](https://docs.sentry.io/platforms/javascript/session-replay/#lazy-loading-replay).
Does `integrations` in `Sentry.init` have to be an empty array? to use `addIntegration`?

```typescript
Sentry.init({
  // Note, Replay is NOT instantiated below:
  integrations: [],
});

// Sometime later
const { Replay } = await import("@sentry/browser");
getCurrentHub()
  .getClient()
  .addIntegration(new Replay());
```

Reply:
Because I already use the integrations to use `BrowserTracing` like this:

```typescript
Sentry.init({
  // Note, Replay is NOT instantiated below:
  integrations: [new Integrations.BrowserTracing()],
});

// Sometime later
const { Replay } = await import("@sentry/browser");
getCurrentHub()
  .getClient()
  .addIntegration(new Replay());
```

I just wonder I can use both integrations.